compiler/hash-pest-parser/src/translate.rs

```diff
@@ -586,23 +587,22 @@ where
                         let mut components = inner.into_inner();
                         let num = components.next().unwrap();
 
-                        let val = BigInt::parse_bytes(num.as_str().as_bytes(), 10).unwrap();
-
+                        let val = u64::from_str_radix(num.as_str(), 10)?;
                         Ok(ab.node(Literal::Int(val)))
                     }
                     Rule::hex_literal => {
                         let item = inner.into_inner().next().unwrap();
-                        let val = BigInt::parse_bytes(item.as_str().as_bytes(), 16).unwrap();
+                        let val = u64::from_str_radix(item.as_str(), 16)?;
                         Ok(ab.node(Literal::Int(val)))
                     }
                     Rule::octal_literal => {
                         let item = inner.into_inner().next().unwrap();
-                        let val = BigInt::parse_bytes(item.as_str().as_bytes(), 8).unwrap();
+                        let val = u64::from_str_radix(item.as_str(), 8)?;
                         Ok(ab.node(Literal::Int(val)))
                     }
                     Rule::bin_literal => {
                         let item = inner.into_inner().next().unwrap();
-                        let val = BigInt::parse_bytes(item.as_str().as_bytes(), 2).unwrap();
+                        let val = u64::from_str_radix(item.as_str(), 2)?;
                         Ok(ab.node(Literal::Int(val)))
```

Comment:
Once we merge the `tokeniser` stuff and decide on a formal spec for float and int literals, we should move away from relying on letting Rust parse these for us. Since we want to control the behaviour and not let it be guided by an external format.

Reply:
Agreed.